### ISSUE TITLE
Bug: nil type assertion handling

### ISSUE THREAD

**Issue by ccoVeille** [polyfloyd/go-errorlint#101]
Please consider the following code

```go
// package foo blah
package foo

// Foo is blah
func Foo() {
	// we call a function and we expect an error
	err := getFromDB()
	switch err.(type) {
	case *FoundError: // this is the expected error
		// do nothing, this is expected
		return
	case nil:
		// we should have an error
		panic("expected an error, but got nil")
	}

	// proceed with normal execution
}

// FoundError is a custom error type
type FoundError struct{}

func (*FoundError) Error() string {
	return "an error occurred"
}

func getFromDB() error {
	return nil
}
```

errorlint will fix it like this

```go
// package foo blah
package foo

import "errors"

// Foo is blah
func Foo() {
	// we call a function and we expect an error
	err := getFromDB()
	{
		var errCase0 *FoundError
		var errCase1 nil
		switch {
		case errors.As(err, &errCase0):
			return
		case errors.As(err, &errCase1):
			panic("expected an error, but got nil")
		}
	}

	// proceed with normal execution
}

// FoundError is a custom error type
type FoundError struct{}

func (*FoundError) Error() string {
	return "an error occurred"
}

func getFromDB() error {
	return nil
}
```

Which is invalid, because nil is not a type

I bumped on it by applying --fix on a large code base at work.
I provided a minimal reproducible issue, here the code may sound strange, but we had this pattern in the code.

Here is what could be expected as a fix

```go
var errCase *FoundError
switch {
	case errors.As(err, &errCase):
		return
	case err == nil:
		panic("expected an error, but got nil")
}
```

**Comment by ccoVeille** [comment]
Does the fix suppprt something like

```
var errCase *DuplicatedError
switch {
	case errors.As(err, &errCase):
		// we can ignore this error
	case err != nil:
		return err
}

// no error, yeah
```

I saw the commit check for EQL, but is there something that check for != nil @polyfloyd

**Comment by polyfloyd** [comment]
I only fixed the rewriting part of the code. I did not test with such a case. But there should be logic already to ignore such comparisons, although I can not remember anymore whether this was also applied to switch statements

**Comment by ccoVeille** [comment]
I checked and everything is fine indeed.
